ls1intum/Artemis
# `Iris`: Fix two UX issues

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


There were three minor issues with the client code of Iris:
1. The resend feature did not work
2. If certain errors happened, it was impossible to start a new session
3. The tests did not pass

### Description

This PR fixes all three issues:
1. Resend now works again in all relevant cases
2. You can now always start a new session
3. The tests pass

### Steps for Testing

(Don't be scared by the length, they are just very detailed 😉)

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Navigate to [exercise 1 in course 1](https://artemis-test9.artemis.cit.tum.de/courses/1/exercises/1) (student view)
3. Start the exercise (if not already done previously)
4. Wait for it to be started
5. Open Iris by clicking on the speech bubble in the bottom right
6. Send a message, wait for a response -> works
7. Now close Iris, click the "View" button and then click the "Iris" button
8. Change the first `{{#system~}}` to `{{system~}}`
9. Go back to the student view
10. Send a message -> Error occurs
11. You can see the resend symbol -> click it -> error again obviously
12. You can start a new session by clicking on the trash can
13. Send a message -> does not work
14. **Important: Change the `{{system~}}` to `{{#system~}}`!**
15. Resend the message -> works

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage







### Screenshots
